src/test/java/com/google/maps/PlacesApiIntegrationTest.java

```diff
@@ -0,0 +1,182 @@
+
+    // Name
+    assertNotNull(placeDetails.name);
+    assertEquals(placeDetails.name, "Google");
+
+    // Opening Hours
+    assertNotNull(placeDetails.openingHours);
+    assertNotNull(placeDetails.openingHours.openNow);
+    assertNotNull(placeDetails.openingHours.periods);
+    assertNotNull(placeDetails.openingHours.weekdayText);
+    assertNotNull(placeDetails.utcOffset);
+
+    // Sydney can be either UTC+10 or UTC+11
+    assertTrue(placeDetails.utcOffset == 600 || placeDetails.utcOffset == 660);
+
+    // Photos
+    {
+      assertNotNull(placeDetails.photos);
+      Photo photo = placeDetails.photos[0];
+      assertNotNull(photo);
+      assertNotNull(photo.photoReference);
+      assertNotNull(photo.height);
+      assertNotNull(photo.width);
+      assertNotNull(photo.htmlAttributions);
+      assertNotNull(photo.htmlAttributions[0]);
+    }
+
+    // Reviews
+    {
+      assertNotNull(placeDetails.reviews);
+      PlaceDetails.Review review = placeDetails.reviews[0];
+      assertNotNull(review);
+      assertNotNull(review.authorName);
+      assertNotNull(review.authorUrl);
+      assertNotNull(review.language);
+      assertNotNull(review.rating);
+      assertNotNull(review.aspects);
+      PlaceDetails.Review.AspectRating aspect = review.aspects[0];
+      assertNotNull(aspect);
+      assertNotNull(aspect.rating);
+      assertNotNull(aspect.type);
+    }
+    // Place ID
+    assertNotNull(placeDetails.placeId);
+    assertEquals(placeDetails.placeId, GOOGLE_SYDNEY);
+    assertNotNull(placeDetails.scope);
+    assertEquals(placeDetails.scope, PlaceIdScope.GOOGLE);
+    assertNotNull(placeDetails.types);
+    assertEquals(placeDetails.types[0], "establishment");
+    assertNotNull(placeDetails.rating);
+    assertEquals(placeDetails.rating, 4.4, 1.0);
+    assertNotNull(placeDetails.userRatingsTotal);
+    assertTrue(placeDetails.userRatingsTotal > 95);
+  }
+
+  @Test
+  public void testTextSearch() throws Exception {
+    PlacesSearchResponse response = PlacesApi.textSearchQuery(context, "Google Sydney").await();
+
+    assertNotNull(response);
+
+    assertNotNull(response.results);
+    assertEquals(1, response.results.length);
+    {
+      PlacesSearchResult result = response.results[0];
+      assertNotNull(result);
+      assertNotNull(result.formattedAddress);
+      assertEquals("5, 48 Pirrama Rd, Pyrmont NSW 2009, Australia", result.formattedAddress);
+      assertNotNull(result.placeId);
+      assertEquals("ChIJN1t_tDeuEmsRUsoyG83frY4", result.placeId);
+    }
+
+  }
+
+  @Test
+  public void testPizzaInNewYork() throws Exception {
+    PlacesSearchResponse response = PlacesApi.textSearchQuery(context, "Pizza in New York").await();
+    assertNotNull(response);
+    assertNotNull(response.results);
+    assertEquals(20, response.results.length);
+    assertNotNull(response.nextPageToken);
+
+    // The returned page token is not valid for a couple of seconds.
```

Comment:
The nextPageToken javadoc contains the following:

There is a short delay between when this response is issued, and when nextPageToken will become valid to execute.

I'll pull it out as a note on a separate line, but do you feel we need to say more here?